Flattens an image's transparencies (GD/ImageMagick) & ImageMagick CLI bug fix

JakeAi opened the pull request:
Added a function that flattens an image's transparencies (PNG > JPG) and defaults to a white background instead of black (similar to Photoshop) and fixes ImageMagick CLI in Windows (at least Win10/8/2012/2012r2) escaping.

JakeAi commented:
Default white background
```PHP
$imgHandler = \Config\Services::image()
			->withFile( "{$folderPath}/{$this->filename}.{$this->extension}" )
			->flatten()
			->crop( $width, $height, $xOffset, $yOffset )
			->resize( 500, 500 )
			->save( "{$folderPath}/thumb_{$this->filename}.jpg" );

```

Red background
```PHP
$imgHandler = \Config\Services::image()
			->withFile( "{$folderPath}/{$this->filename}.{$this->extension}" )
			->flatten(255, 0, 0)
			->crop( $width, $height, $xOffset, $yOffset )
			->resize( 500, 500 )
			->save( "{$folderPath}/thumb_{$this->filename}.jpg" );

```

lonnieezell commented on system/Images/Handlers/ImageMagickHandler.php:
```diff
@@ -72,7 +72,14 @@ public function _resize(bool $maintainRatio = false)
 		$destination = $this->getResourcePath();
 
-		$action = $maintainRatio === true ? ' -resize ' . $this->width . 'x' . $this->height . ' "' . $source . '" "' . $destination . '"' : ' -resize ' . $this->width . 'x' . $this->height . '\! "' . $source . '" "' . $destination . '"';
+		//todo FIX THIS HANDLER PROPERLY
```
What's needed to fix this properly?

JakeAi replied:
Making sure it works on windows 7 and other OS. Windows (10/2012) doesn't like the \! For whatever reason. I just don't have access to other OS and not really enough time since I'm using this as our platform! :)

JakeAi commented:
Well, I'm also a graphic designer and in every adobe program it's been called "flatten transparencies" because of the image type not supporting transparency. Transparencies need flattened into pixels with no Alpha channel so a black with 50% alpha (0,0,0,50) would actually turn into a gray

JakeAi commented:
I can work on docs I guess. I just have to figure that out yet. I finally figured out signing GitHub commits before posting this lol. I was using it to upload an image and create 10 diff size thimbnails but GD was giving us color banding. So I switched to imagemagick and had better results, once I fixed it. Then I was getting black when saving to jpg. So I implemented flatten. And now it's solid!

JakeAi commented:
Also, in imagemagick it's called flatten as well... forgot about that.

lonnieezell commented:
Ah, that makes sense on the naming then. I'm cool with it. Docs are pretty simple. They're in /user_guide_src/source and are Restructured Text files.

Thanks!

JakeAi commented:
Added documentation, I don't know if I need to do anything else to the RST. I've never created them before.

lonnieezell commented:
Looks good. Thanks!
